presto-feature-toggle/src/main/java/com/facebook/presto/features/binder/FeatureToggleBinder.java

```diff
@@ -0,0 +1,177 @@
+import static java.util.Objects.requireNonNull;
+
+public class FeatureToggleBinder<T>
+{
+    private final Binder binder;
+    private Class<T> baseClass;
+    private Class<? extends T> defaultClass;
+    private Set<Class<? extends T>> classes = new HashSet<>();
+    private boolean hotReloadable;
+    private String strategy;
+    private String featureId;
+    private boolean enabled = true;
+    private Map<String, String> strategyConfigMap = new ConcurrentHashMap<>();
+
+    public FeatureToggleBinder(Binder binder, Class<T> baseClass, Class<? extends T> defaultClass, Set<Class<? extends T>> classes, boolean hotReloadable, String strategy, String featureId, boolean enabled, Map<String, String> strategyConfigMap)
+    {
+        this.binder = binder;
+        this.baseClass = baseClass;
+        this.defaultClass = defaultClass;
+        this.classes = classes;
+        this.hotReloadable = hotReloadable;
+        this.strategy = strategy;
+        this.featureId = featureId;
+        this.enabled = enabled;
+        this.strategyConfigMap = strategyConfigMap;
+    }
+
+    private FeatureToggleBinder(Binder binder)
+    {
+        this.binder = requireNonNull(binder, "binder is null").skipSources(getClass());
+    }
+
+    public FeatureToggleBinder(Binder binder, Class<T> klass)
+    {
+        this.binder = requireNonNull(binder, "binder is null").skipSources(getClass());
+        this.baseClass = klass;
+    }
+
+    public static <T> FeatureToggleBinder<T> featureToggleBinder(Binder binder, Class<T> klass)
+    {
+        return new FeatureToggleBinder<>(binder, klass);
+    }
+
+    public static <T> FeatureToggleBinder<T> featureToggleBinder(Binder binder)
+    {
+        return new FeatureToggleBinder<>(binder);
+    }
+
+    public FeatureToggleBinder<T> baseClass(@NotNull Class<T> baseClass)
+    {
+        return new FeatureToggleBinder<>(binder, baseClass, defaultClass, classes, hotReloadable, strategy, featureId, enabled, strategyConfigMap);
+    }
+
+    public FeatureToggleBinder<T> featureId(@NotNull String featureId)
+    {
+        this.featureId = featureId;
+        return this;
+    }
+
+    public FeatureToggleBinder<T> defaultClass(@NotNull Class<? extends T> defaultClass)
+    {
+        this.defaultClass = defaultClass;
+        this.classes.add(defaultClass);
+        return this;
+    }
+
+    @SafeVarargs
+    public final FeatureToggleBinder<T> allOf(@NotNull Class<? extends T>... classes)
+    {
+        this.classes = new HashSet<>(Arrays.asList(classes));
+        return this;
+    }
+
+    public FeatureToggleBinder<T> enabled(boolean enabled)
+    {
+        this.enabled = enabled;
+        return this;
+    }
+
+    public FeatureToggleBinder<T> toggleStrategy(@NotNull String strategy)
+    {
+        this.strategy = strategy;
+        return this;
+    }
+
+    public FeatureToggleBinder<T> toggleStrategyConfig(@NotNull Map<String, String> strategyConfigMap)
```

Comment:
Add explicit null checks.  Please do this everywhere where you've used the @NotNull annotation.

Reply:
done